diff --git a/faiss/impl/ScalarQuantizer.cpp b/faiss/impl/ScalarQuantizer.cpp
index 07d77d56..77f7d649 100644
--- a/faiss/impl/ScalarQuantizer.cpp
+++ b/faiss/impl/ScalarQuantizer.cpp
@@ -52,6 +52,15 @@ namespace faiss {
 #endif
 #endif
 
+#if defined(__aarch64__)
+#if defined(__GNUC__) && __GNUC__ < 8
+#warning \
+        "Cannot enable NEON optimizations in scalar quantizer if the compiler is GCC<8"
+#endif
+#else
+#define USE_NEON
+#endif
+
 namespace {
 
 typedef ScalarQuantizer::QuantizerType QuantizerType;
